build/ci/job-template.yml

```diff
@@ -62,6 +62,15 @@ jobs:
         displayName: List latest package versions
       - script: $(dotnetPath) run --project $(packageUpdaterProjPath)
         displayName: Update package versions for nightly build
+      - ${{ if eq(parameters.buildScript, 'build.cmd') }}:
+        - powershell: |
```

Comment:
Can you please explain why this is useless? If this useless, why do we have it in our tree and have it build?
If the reason is different, can you please add the actual reason? (Also, small typo: useless)

Reply:
Yes, for NightlyBuild pipeline only NightlyBuild test project is necessary, other test projects are built from previous "Build" step. This "Build" are doing multiple things thatNightlyBuild pipeline also requires, like initialize tools, build native components. We are reusing the "Build" step from CI build pipeline but with side effect that we are also building too much test projects. We can separate the build step into smaller steps and only include necessary steps in nightly build pipeline but that can be quite large effort and I'm not seeing much value of this.

And thanks for pointing out the typo, I will fix the typo

---
In reply to: [391391849](https://github.com/dotnet/machinelearning/pull/4935#discussion_r391391849) [](ancestors = 391391849)